cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -134,13 +142,14 @@ func MustNew(cfg config.LocalConfig) *Daemon {
 
 	handler, err := internal.NewJSONRPCHandler(internal.HandlerParams{
 		AccountStore:      methods.AccountStore{Client: hc},
-		EventStore:        methods.EventStore{Client: hc},
+		EventStore:        eventStore,
 		FriendbotURL:      cfg.FriendbotURL,
 		NetworkPassphrase: cfg.NetworkPassphrase,
 		Logger:            logger,
 		TransactionProxy:  transactionProxy,
 		CoreClient:        &stellarcore.Client{URL: cfg.StellarCoreURL},
 		LedgerEntryReader: db.NewLedgerEntryReader(dbConn),
+		MaxEventsLimit:    cfg.MaxEventsLimit,
```

Comment:
hmm. we might want to pass the entire `*cfg` instead of specifying specific fields. That would save us from adding fields over an over. but, that's for another PR.